lib/linters.py

```diff
@@ -0,0 +1,627 @@
+      config = yaml.load(f.read())
+  else:
+    config = {}
+  linters = [kls(errors=errors, config=config) for kls in get_linters()]
+  return (errors, linters)
+
+
+class BaseLintRule(object):
+  """Base Lint Rule class. All lint rules should extend this.
+
+  All configurable parameters should be placed into DEFAULTS - as these
+  values get merged with the optional user-specified YAML file.
+
+  setup() can be implemented by child classes that might need to do
+  additional configuration such as remote data-fetching.
+  """
+
+  CONFIG_NAME = None
+  DEFAULT_ENABLED = True
+  CATEGORY = None
+  DEFAULTS = {}
+
+  def __init__(self, errors, config):
+    self.global_config = config
+    self.errors = errors
+    # build a local copy of the linter specific config
+    cn = self.CONFIG_NAME or self.__class__.__name__
+    local_config = deepcopy(self.DEFAULTS)
+    local_config.update(self.global_config.get(cn, {}))
+    self.config = local_config
+    self.setup()
+
+  def setup(self):
+    """setup() may optionally be implemented by each linter"""
+    pass
+
+  def add(self, *args, **kwargs):
+    if 'category' not in kwargs and self.CATEGORY:
+      kwargs['category'] = self.CATEGORY
+    self.errors.add(*args, **kwargs)
+
+  def lint_naming(self, definitions):
+    """Execute the lint check_* methods against the specified definitions."""
+    if not self.config.get('enabled', self.DEFAULT_ENABLED):
+      return
+
+    for itemunit in definitions.networks.values():
+      self.check_network(itemunit)
+
+    for itemunit in definitions.services.values():
+      self.check_service(itemunit)
+
+    self.check_networks(definitions.networks)
+    self.check_services(definitions.services)
+
+  def lint_policy(self, policy):
+    """Execute the lint check_term method against the specified policy"""
+    if not self.config.get('enabled', self.DEFAULT_ENABLED):
+      return
+
+    for header, terms in policy.filters:
+      if not terms:
+        # no terms included -probably just include statements
+        continue
+      for term in terms:
+        self.check_term(policy.filename, header, term)
+
+  def check_network(self, network):
+    """May be implemented to check a given network found in a .net file"""
+    pass
+
+  def check_service(self, service):
+    """May be implemented to check a given service found in a .svc file"""
+    pass
+
+  def check_networks(self, networks):
+    """May be implemented to check all of the networks found in a .net file"""
+    pass
+
+  def check_services(self, services):
+    """May be implemented to check all of the services found in a .svc file"""
+    pass
+
+  def check_term(self, policy, header, term):
+    """May be implemented to check the given term under a given header"""
+    pass
+
+
+@register_linter
+class AddressNumEnforce(BaseLintRule):
+  """Warns about terms with more than a few address tokens."""
+
+  def check_term(self, policy, header, term):
+    addr_attrs = (
+      'source_address',
```

Comment:
Add two spaces on this and the next couple lines to bring them to the correct indentation.